src/main/java/cloud/orbit/spring/actuate/ActorInfoContributorLifetimeExtension.java

```diff
@@ -0,0 +1,176 @@
+
+    ActorInfoContributorLifetimeExtension(final OrbitRemoteReferenceService orbitRemoteReferenceService,
+                                          final LifetimeExtension springLifetimeExtension)
+    {
+        this.orbitRemoteReferenceService = orbitRemoteReferenceService;
+        this.springLifetimeExtension = springLifetimeExtension;
+    }
+
+    @Override
+    public <T> T newInstance(final Class<T> concreteClass)
+    {
+        // In case Orbit chooses this extension as the "primary" LifetimeExtension, we need to redirect to the
+        // SpringLifetimeExtension for constructing new instances
+        return springLifetimeExtension.newInstance(concreteClass);
+    }
+
+    @Override
+    public Task<?> preActivation(final AbstractActor<?> actor)
+    {
+        return doIfInfoContributor(actor, actorInfoContributorReferences::add);
+    }
+
+    @Override
+    public Task<?> preDeactivation(final AbstractActor<?> actor)
+    {
+        return doIfInfoContributor(actor, actorInfoContributorReferences::remove);
+    }
+
+    private Task<?> doIfInfoContributor(final AbstractActor<?> actor,
+                                        final Consumer<ActorInfoContributorReference> synchronizedAction)
+    {
+        if (actor instanceof InfoContributor)
+        {
+            ActorInfoContributorReference actorInfoContributorReference = new ActorInfoContributorReference(actor);
+            synchronized (actorInfoContributorReferences)
+            {
+                synchronizedAction.accept(actorInfoContributorReference);
+            }
+        }
+        return Task.done();
+    }
+
+    @Override
+    public void contribute(final Info.Builder builder)
+    {
+        createActorInfoObject().ifPresent(actorInfoObject -> builder.withDetail("actors", actorInfoObject));
+    }
+
+    private Optional<?> createActorInfoObject()
+    {
+        Set<ActorInfoContributorReference> actorInfoContributorReferencesCopy;
+        synchronized (actorInfoContributorReferences)
+        {
+            actorInfoContributorReferencesCopy = actorInfoContributorReferences.stream()
+                    .map(ActorInfoContributorReference::new)
+                    .collect(Collectors.toSet());
+        }
+        Collection<ActorInfoContributorReference> expiredReferences = new ArrayList<>();
+        Map<String, Map<String, Object>> actorInfoObject = new HashMap<>();
+        actorInfoContributorReferencesCopy.forEach(actor ->
+        {
+            InfoContributor contributor = actor.reference.get();
+            if (contributor == null)
+            {
+                expiredReferences.add(actor);
+            }
+            else
+            {
+                final Info.Builder builder = new Info.Builder();
+                contributor.contribute(builder);
+                actorInfoObject.putIfAbsent(actor.name, new HashMap<>());
+                actorInfoObject.get(actor.name).put(actor.identity, builder.build().getDetails());
```

Comment:
Config proposal:
```yaml
management.info.actors.group.primary: NONE (default) | INTERFACE | IDENTITY
management.info.actors.group.secondary: NONE (default) | INTERFACE | IDENTITY
```
So it defaults to the way InfoContributor is expected to behave and allows customization of how the info is grouped. The structure I was advocating earlier would be configured like so:
```yaml
management.info.actors.group.primary: INTERFACE
management.info.actors.group.secondary: IDENTITY
```